jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java

```
@@ -174,6 +175,20 @@ public SubProcessNode callActivity(long id, String name, String calledId, boolea
         subProcessNode.setName(name);
         subProcessNode.setProcessId(calledId);
         subProcessNode.setWaitForCompletion(waitForCompletion);
+        subProcessNode.setIndependent(true);
+
+        VariableScope variableScope = new VariableScope();
+        subProcessNode.addContext(variableScope);
+        subProcessNode.setDefaultContext(variableScope);
+
+        Map<String, String> inputOtuputTypes = new HashMap<>();
+        inputOtuputTypes.put(DEFAULT_WORKFLOW_VAR, JSON_NODE);
+        subProcessNode.setMetaData("BPMN.InputTypes", inputOtuputTypes);
+        subProcessNode.setMetaData("BPMN.OutputTypes", inputOtuputTypes);
+
+        // parent and sub processes have process var "workflowdata"
```

Probably would be better to omit it as if the value of the `DEFAULT_WORKFLOW_VAR` is changed, this comment becomes misleading.

---

good point - fixed